src/routes/(app)/expenses/[id]/+page.server.ts

```diff
@@ -0,0 +1,230 @@
+      throw error(404, "Expense id should be a number");
+
+    if (!form.valid) return fail(400, { form });
+
+    const { items, ...expenseData } = form.data;
+
+    const expense = await prisma.expense.update({
+      where: { id: Number(params.id) },
+      data: {
+        date: expenseData.date,
+        description: expenseData.description,
+        isGuildCard: expenseData.isGuildCard,
+      },
+    });
+
+    const newItems = items.filter((item) => !item.itemId);
+    const existingItems = items.filter((item) => !!item.itemId);
+
+    try {
+      await prisma.expenseItem.deleteMany({
+        where: {
+          expenseId: Number(params.id),
+          id: {
+            notIn: existingItems.map((item) => item.itemId!),
+          },
+        },
+      });
+    } catch (e) {
+      if (e instanceof Prisma.PrismaClientKnownRequestError) {
+        if (e.code === "P2004") {
+          // can't delete due to access
+        }
+      } else {
+        throw e;
+      }
+    }
+
+    // Update the related expense items
+    if (existingItems.length > 0) {
+      const updateItemsPromises = await Promise.allSettled(
+        existingItems.map((item) =>
+          prisma.expenseItem.update({
+            where: { id: item.itemId! },
+            data: {
+              costCenter: item.costCenter,
+              committeeShortName: getCostCenter(item.costCenter).committee,
+              amount: item.amount,
+              comment: item.comment,
+            },
+          }),
+        ),
+      );
+      for (const promiseResult of updateItemsPromises)
+        if (promiseResult.status === "rejected") {
+          console.error(`Failed to update expense item`, promiseResult.reason);
+          throw promiseResult.reason;
+        }
+    }
+    if (newItems.length > 0) {
+      await updateSignersCacheIfNecessary();
+      const newItemsPromises = await Promise.allSettled(
+        newItems.map((item) =>
+          prisma.expenseItem.create({
+            data: {
+              expenseId: Number(params.id),
+              costCenter: item.costCenter,
+              committeeShortName: getCostCenter(item.costCenter).committee,
+              amount: item.amount,
+              comment: item.comment,
+              signerMemberId: resolveSignerLogic(
+                getSigner(item.costCenter),
+                expense.memberId, // user who created the expense
+                item.costCenter,
+              ),
+            },
+          }),
+        ),
+      );
+      for (const promiseResult of newItemsPromises)
+        if (promiseResult.status === "rejected") {
+          console.error(`Failed to create expense item`, promiseResult.reason);
+          throw promiseResult.reason;
+        }
+    }
+
+    return message(form, {
+      message: "Utlägg uppdaterat",
+      type: "success",
+    });
+  },
+  delete: async (event) => {
+    const { locals, params } = event;
+    const { prisma } = locals;
+    if (Number.isNaN(Number(params.id)))
```

Comment:
`Number.isNaN(Number('')) === false` är en grej som står ut till mig när man vill kolla om ngt är ett nummer i js; mer att jag hade velat faila i det fallet oxå. kanske orimligt/nit med tanke på hur nummer funkar i javascript men lämnar det upp till dig.

Reply:
Jag tror det är okej. Enda anledningen jag gör detta är att om vi skickar in `id: "hej"` till prisma får vi ett database validation error, vilket visar på felaktig kod (eftersom vi skickar in en sträng till ett nummerfält). Om vi däremot skickar in `id: 0` får vi en vanlig 404, vilket är det vi vill. Så om `id = ''` borde väl det bli `id: 0`?

Reply:
Det blir id: 0, tror min tanke snarare var att om någon har sönder något, så en tom-sträng skickas in alltid, borde det gå sönder här så man märker det enklare

Reply:
fair point